cmd/lotus-shed/actor.go

```diff
@@ -159,6 +160,78 @@ var actorWithdrawCmd = &cli.Command{
 	},
 }
 
+var beneficiaryWithdrawCmd = &cli.Command{
+	Name:      "withdraw",
+	Usage:     "withdraw available balance to beneficiary address. Will not work from non-beneficiary owner addresses. Use lotus-miner for non-beneficiary owner withdrawals",
+	ArgsUsage: "[minerAddress, amount (FIL)]",
+	Flags: []cli.Flag{
+		&cli.IntFlag{
+			Name:  "confidence",
+			Usage: "number of block confirmations to wait for",
+			Value: int(build.MessageConfidence),
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		ctx := lcli.ReqContext(cctx)
+		if cctx.Args().Len() != 2 {
+			return xerrors.Errorf("expected 2 arguments")
+		}
+
+		maddr, err := address.NewFromString(cctx.Args().Get(0))
+		if err != nil {
+			return fmt.Errorf("parsing address %s: %w", maddr, err)
+		}
+
+		amt, err := types.ParseFIL(cctx.Args().Get(1))
+		if err != nil {
+			return xerrors.Errorf("parsing 'amount' argument: %w", err)
+		}
+
+		amount := abi.TokenAmount(amt)
+
+		nodeAPI, acloser, err := lcli.GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer acloser()
+
+		minerAPI, closer, err := lcli.GetStorageMinerAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		msgCid, err := minerAPI.BeneficiaryWithdrawBalance(ctx, maddr, amount)
+		if err != nil {
+			return err
+		}
```

Comment:
@arajasek I imagine that I might have to used `MpoolPushMessage` here instead of relying on the miner-api that might not be present?

Reply:
Yes, see the command directly above this one. Honestly, you could fold this command into that one -- just add a param for the user to indicate whether to send from owner or beneficiary. They're otherwise identical.